optaplanner-benchmarks/optaplanner-perf-benchmark/src/main/java/org/jboss/qa/brms/performance/micro/AbstractPlannerMoveMicroBenchmark.java

```diff
@@ -0,0 +1,51 @@
+package org.jboss.qa.brms.performance.micro;
+
+import java.util.concurrent.TimeUnit;
+
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.optaplanner.core.impl.heuristic.move.Move;
+import org.optaplanner.core.impl.score.director.ScoreDirector;
+
+@State(Scope.Thread)
+@BenchmarkMode(Mode.SampleTime)
```

Comment:
It would be better to use AverageTime. It is similar as SampleTime, however taking into account all measured values.  

Edit: Looking at the later code, I am also not sure about AverageTime. It depends if doMove() and rebase() operate always over the same set of data at the same state.

Reply:
The said operations do actually change the data set they're working on, but it was fixed by setting `Setup` method to `Level.Invocation`. Thanks for suggestion! :+1: